java/plugins/web-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/WebElementsPage.java

```diff
@@ -0,0 +1,497 @@
+     * space-trimmed before assertion, so only non-space characters will matter
+     *
+     * @param text text to search for
+     */
+    @ActionTitle("ru.sbtqa.tag.pagefactory.text.absent.on.page")
+    public void assertTextIsNotPresent(String text) {
+        WebExtension.waitForTextPresenceInPageSource(text, false);
+    }
+
+    /**
+     * Wait for a new browser window, then wait for a specific text inside the
+     * appeared window List of previously opened windows is being saved before
+     * each click, so if modal window appears without click, this method won't
+     * catch it. Text is being waited by {@link #assertTextAppears}, so it will
+     * be space-trimmed as well
+     *
+     * @param text text that will be searched inside of the window
+     * @throws ru.sbtqa.tag.pagefactory.exceptions.WaitException if
+     */
+    @ActionTitle("ru.sbtqa.tag.pagefactory.modal.window.with.text.appears")
+    public void assertModalWindowAppears(String text) throws WaitException {
+        try {
+            String popupHandle = WebExtension.findNewWindowHandle((Set<String>) Stash.getValue("beforeClickHandles"));
+            if (null != popupHandle && !popupHandle.isEmpty()) {
+                PageFactory.getWebDriver().switchTo().window(popupHandle);
+            }
+            assertTextAppears(text);
+        } catch (Exception ex) {
+            throw new WaitException("Modal window with text '" + text + "' didn't appear during timeout", ex);
+        }
+    }
+
+    /**
+     * Perform {@link #checkValue(String, WebElement, MatchStrategy)} for the
+     * WebElement with corresponding title on a current page. Use exact match
+     * strategy
+     *
+     * @param text string value that will be searched inside of the element
+     * @param elementTitle title of the element to search
+     * @throws ru.sbtqa.tag.pagefactory.exceptions.ElementNotFoundException if
+     * couldn't find element by given title, or current page isn't initialized
+     */
+    @ActionTitle("ru.sbtqa.tag.pagefactory.check.value")
+    public void checkValue(String elementTitle, String text) throws PageException {
+        checkValue(text, PageFactoryUtils.getElementByTitle(PageContext.getCurrentPage(), elementTitle), MatchStrategy.EXACT);
+    }
+
+    /**
+     * Perform {@link #checkValue(String, WebElement, MatchStrategy)} for the
+     * specified WebElement. Use exact match strategy
+     *
+     * @param text string value that will be searched inside of the element
+     * @param webElement WebElement to check
+     */
+    public void checkValue(String text, WebElement webElement) {
+        checkValue(text, webElement, MatchStrategy.EXACT);
+    }
+
+    /**
+     * Define a type of given WebElement, and check whether it either contains,
+     * or exactly matches given text in its value. Currently supported elements
+     * are text input and select box TODO: use HtmlElements here, to define
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)